File: algorithms/javascript/bubble-sort.js
function bubbleSort(nums) {
  var swapped = true;
  while(swapped) {
    swapped = false;
    for (var i = 0; i < nums.length; i++) {
      if(nums[i] > nums[i+1]) {
        let tmp = nums[i];
        nums[i] = nums[i+1];
        nums[i+1] = tmp;
        swapped = true;
      }
    }
  }
  return nums;
}

module.exports = bubbleSort;

File: algorithms/ruby/app/bubble_sort.rb
class BubbleSort
  attr_accessor :array, :size

  def initialize(array)
    @array = array
    @size = array.size
  end

  def sort
    return array if size < 2

    swapped = true

    while swapped do
      swapped = false

      (size - 1).times do |n|
        if array[n] > array[n+1]
          array[n], array[n+1] = array[n+1], array[n]
          swapped = true
        end
      end
    end

    array
  end
end
